src/Extension/CoreExtension.php

```diff
@@ -766,9 +766,9 @@ public static function last(string $charset, $item)
      *  {{ [1, 2, 3]|join }}
      *  {# returns 123 #}
      *
-     * @param array       $value An array
-     * @param string      $glue  The separator
-     * @param string|null $and   The separator for the last pair
+     * @param iterable|array|string|float|int|bool|object|null  $value An array
```

Comment:
`object` should probably be `Stringable` instead, as non-stringable objects won't work in `implode`

Reply:
It is casted to array, https://3v4l.org/FqgLS

Reply:
oh. This is probably unexpected behavior.

Reply:
So what to do? 

/cc @fabpot

Reply:
I would suggest not including `object` in the type here (without changing the implementation for now). This would make TwigStan report cases triggering this weird behavior as mistakes.
I'm not sure about the scalar types, for which the casting is more expected.

Reply:
Good idea, done!